src/libraries/System.Private.CoreLib/src/System/Collections/Generic/HashSet.cs

```diff
@@ -1007,39 +982,52 @@ private void Resize(int newSize, bool forceNewHashCodes)
         /// </summary>
         public void TrimExcess()
         {
-            int capacity = Count;
-
-            int newSize = HashHelpers.GetPrime(capacity);
-            Entry[]? oldEntries = _entries;
-            int currentCapacity = oldEntries == null ? 0 : oldEntries.Length;
-            if (newSize >= currentCapacity)
+            int newCapacity = HashHelpers.GetPrime(_count);
```

Comment:
This looks like an unrelated change?

Reply:
I changed Construct from method, as it always expanded prime to be larger than original collection